lib/analyzer/image-inspector.ts

```diff
@@ -39,17 +55,21 @@ async function detect(
  */
 async function getImageArchive(
   targetImage: string,
+  imageSavePath: string,
   username?: string,
   password?: string,
 ): Promise<ArchiveResult> {
   const docker = new Docker(targetImage);
-  const destination = tmp.dirSync({ unsafeCleanup: true });
+  mkdirp.sync(imageSavePath);
+  const destination: DestinationDir = {
+    name: imageSavePath,
+    removeCallback: cleanupCallback(imageSavePath, "image.tar"),
```

Comment:
that's not the name of the image now, it doesn't delete the right file.
this also points to a missing test :)

Reply:
Thanks to show me it! :)

Reply:
I was reviewing the solution and it's better that the image name keeps that same, because on `snyk-docker-pull` it uses `image.tar`. And there is no way to customaise it, unless we make another change there.
However I got your idea to create a tmp folder in the path and save the image there. I think it has the same effect. Unless I'm missing something.

Reply:
it might could have been better to make `snyk-docker-pull` to get the full path of the image as a parameter and not only the folder :)
anyway - now we don't delete the folders we create... are we ok with that?

Reply:
I agree with you. That would be better. However we're not ok to not delete the folder. Let's delete it too.